private-templates-service/client/src/components/Common/Tags/Tags.tsx

```diff
@@ -81,12 +83,45 @@ class Tags extends React.Component<Props, State>  {
     e.preventDefault();
     if (this.addTagInput && this.addTagInput.current && this.props.template && this.props.template.tags) {
       const tag = this.addTagInput.current.value;
-      this.props.updateTags([...this.props.template.tags, tag]);
+      if(!this.props.template.tags.includes(tag)){
+        if(tag === ""){
+          this.closeAddTag();
+        }
+        else{
+          this.props.updateTags([...this.props.template.tags, tag]);
+       }
+      }
+      else{
+        this.highlightTag(tag,this.props.template.tags)
+      }
+    }
+  }
+
+  highlightTag = (tagToHighlight: string, currentTags: string[]): void => {
+    for(let tag of currentTags){
+      if(tag === tagToHighlight){
+        if (tag in this.tagRefs) {
+          this.tagRefs[tag].classList.add('duplicate'); // Add the class that renders an animation
+        }
+      }
+    }
+  }
+
+  onAnimationEnd = (event: any) => {
+    if (event && event.target) {
+      event.target.classList.remove('duplicate'); // Once the animation is complete, remove the class from the component
     }
   }
 
   closeAddTag = () => {
     this.setState({ isAdding: false });
+    this.setState({newTagName: ""});
+  }
+
+  onKeyDown = (keyStroke: any) => {
+    if(keyStroke.keyCode === 27){
```

Comment:
I guess it would be better to have some kind of enum to keep keyCodes. It would be more readable, also in the feature if you add more keyCodes the code will still be more readable